crml/nft/src/lib.rs

```diff
@@ -1172,18 +1176,24 @@ impl<T: Config> Module<T> {
 			}
 		}
 		// series schedule takes priority if it exists
-		let mut royalties = Self::series_royalties(bundle_collection_id, bundle_series_id)
-			.unwrap_or_else(|| Self::collection_royalties(bundle_collection_id).unwrap_or_else(Default::default));
+		let mut royalties: RoyaltiesSchedule<T::AccountId> =
+			Self::series_royalties(bundle_collection_id, bundle_series_id).unwrap_or_else(|| {
+				Self::collection_royalties(bundle_collection_id)
+					.unwrap_or_else(|| RoyaltiesSchedule { entitlements: vec![] })
+			});
 		let royalties = match marketplace_id {
 			Some(marketplace_id) => {
 				ensure!(
 					<RegisteredMarketplaces<T>>::contains_key(marketplace_id),
 					Error::<T>::MarketplaceNotRegistered
 				);
 				let marketplace = Self::registered_marketplaces(marketplace_id);
-				royalties
-					.entitlements
-					.push((marketplace.account, marketplace.entitlement));
+				if marketplace.is_some() {
+					let marketplace = marketplace.unwrap();
+					royalties
+						.entitlements
+						.push((marketplace.account, marketplace.entitlement));
+				}
```

Comment:
nit: `if let Some(marketplace) = Self::registered_marketplaces(marketplace_id) {` cleaner for this